### PR TITLE
Fix: update next.js server component example

### PR DESCRIPTION
**Pull request by dijonmusters**
## What kind of change does this PR introduce?

Implement singleton Supabase instance for client components, and restructure steps

## What is the current behavior?

Example contains bugs

## What is the new behavior?

Example contains no bugs 🤞

### CODE REVIEW

**Review by dshukertjr — APPROVED** [review]
LGTM!

**Comment by alaister** [comment]
@dijonmusters would you mind merging in `master` so we can merge? Cheers! 🙏🏻

File: apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx [hunk 1]
```diff
@@ -529,30 +516,33 @@ export default function Login() {
...
 
-import supabase from '../utils/supabase-browser'
+import { useSupabase } from './supabase-provider'
 
+// Supabase auth needs to be triggered client-side
 export default function Login() {
-  const handleLogin = async () => {
-    const { error } = await supabase.auth.signInWithOAuth({
-      provider: 'github',
+  const { supabase, session } = useSupabase()
```
Comment (Chojecki) [inline apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx:523]: Hi, I think provider doesn't expose session anymore

**Comment by dannykng** [comment]
Closing in favor of https://github.com/supabase/supabase/pull/11549.

**Pull request closed without merging**